index.ts

```diff
@@ -86,15 +86,43 @@ export default function scss(options: CSSPluginOptions = {}): Plugin {
                 prev: string,
                 done: ImporterDoneCallback
               ): ImporterReturnType | void => {
-                if (url.startsWith('.')) return null
-
+                // If the import begins with '.', it's local...abort. If it
+                // begins with 'url', 'http', or 'https', it's a remote import,
+                // again abort. Note: we only check for 'http' because both
+                // 'http' and 'https' begin with
+                if (
+                  url.startsWith('.') ||
+                  url.startsWith('url') ||
+                  url.startsWith('http')
+                ) {
+                  return null
+                }
+
+                // If the url begins with '~' remove it. The '~' character is a
+                // standard in the sass community when working in JavaScript
+                // because sass-loader used it as an indicator that the import
+                // was a node_modules import (and still does, I think).
                 const cleanUrl = url.startsWith('~')
                   ? url.replace('~', '')
                   : url
+
+                // Have Node resolve the path to the file we're requesting.
+                // When using a plain node_modules project, this works as
+                // expected. When using Yarn PnP, Yarn will unplug the package,
+                // and give the path to the file. Unplugging a file simply
+                // means unzipping it since most packages aren't equipped to
+                // work with zip archives directly. Also, the 'paths' option
+                // tells Yarn what file is requesting this file, so PnP
+                // integrity remains intact. For Node, paths serves as an
+                // additional set of paths to look for the import in.
                 const resolved = require.resolve(cleanUrl, {
                   paths: [prefix + scss]
                 })
-                return resolved ? { file: resolved } : null
+
+                // Finally, return the resolved file...if we can confirm it
+                // exists. If not, you guessed it...abort, and let sass handle
+                // the import
+                return existsSync(resolved) ? { file: resolved } : null
```

Comment:
`require.resolve` will throw if it can't find it so this should always be true

Reply:
~~I'm not so sure about that one. In my testing, when it couldn't find a file, it just assumed it was relative to the root of the plugin package. It never threw an exception, nor did Yarn (though I know Yarn patches this API for PnP). For example say I was looking for the file `./components/Header/header', if it wasn't able to resolve the path, it just assumed the path was `{PROJECT_ROOT}/.yarn/cache/rollup-plugin-scss-https-ab37d281de-9cbfba33c6.zip/node_modules/rollup-plugin-scss/components/Header/header.scss'. Maybe that's a bug, or maybe it's intended behavior for relative imports, I'm not 100% certain on how the resolution works at the low level.~~

Correction, it is intended behavior. Turns out, the package `resolve-patch` does just that. If the path begins with a relative character, it just does an `fs.resolve()` and assume the path exists. Fair enough, since I've added in all the checks, you are correct, this shouldn't be needed.

Reply:
Given what I now know (sorry, by the way, I'm not very familiar with the internals of Node or Yarn), do you think it's safe to just do `return { file: resolved }`? I mean, I passed local files and urls off to sass, and since `require.resolve()` should throw an error if a file can't be resolved, there's literally no reason the code would get that far if a file wasn't located.

Reply:
Yeah, that should be fine